Source/engine.h

```diff
@@ -598,7 +604,40 @@ void SetRndSeed(int32_t s);
 int32_t AdvanceRndSeed();
 int32_t GetRndSeed();
 int32_t GenerateRnd(int32_t v);
-std::unique_ptr<BYTE[]> LoadFileInMem(const char *pszName, DWORD *pdwFileLen = nullptr);
+
+size_t GetFileSize(const char *pszName);
+void LoadFileData(const char *pszName, byte *buffer, size_t fileLen);
+
+template <typename T, std::size_t N>
+void LoadFileInMem(const char *path, std::array<T, N> &data)
+{
+	LoadFileData(path, (byte *)&data, N * sizeof(T));
+}
+
+/**
+ * @brief Load a file in to a buffer
+ * @param path Path of file
+ * @param pdwFileLen Number of
+ * @return Buffer with content of file
+ */
+template <typename T = byte>
+std::unique_ptr<T[]> LoadFileInMem(const char *path, size_t *elements = nullptr)
+{
+	const size_t fileLen = GetFileSize(path);
+
+	if ((fileLen % sizeof(T)) != 0)
+		app_fatal("File size does not align with type\n%s", path);
+
+	if (elements != nullptr)
+		*elements = fileLen / sizeof(T);
+
+	std::unique_ptr<T[]> buf { new T[fileLen / sizeof(T)] };
+
+	LoadFileData(path, (byte *)buf.get(), fileLen);

```

Comment:
```suggestion
	LoadFileData(path, reinterpret_cast<byte *>(buf.get()), fileLen);
```

Reply:
I went with c-style casting in the places where it needs to be cleaned up and further propagated, just so that it's obvious that it needs to be looked at later.

Reply:
.... but this isn't such a case :D